joomla/joomla-cms
# [5.3]Sort order of fields relative to their actual location in the table

Pull Request for Issue #34501 .

### Summary of Changes
in the element sorting field on the page, the STAGE field is at the very bottom, however, I remember exactly that we previously saved the order in this field relative to the actual location in the table (in this case, the stage line should be the second after ORDERING).



This pull request includes changes to the workflow-related sorting options in the content management system. The changes ensure that these options are only available when the workflow feature is enabled AND that they are positioned in the filter in the correct position

Workflow-related sorting options:

* [`administrator/components/com_content/forms/filter_articles.xml`](diffhunk://#diff-62e20909c83645f0763df23e94695c5084d334ac3cf3f565cf458219248e3e2fR127-R128): Added new sorting options `JSTAGE_ASC` and `JSTAGE_DESC` that require the workflow feature to be enabled.

Conditional form field handling:

* [`administrator/components/com_content/src/Model/ArticlesModel.php`](diffhunk://#diff-add0f2892993b0d9480d645a92e769680f686d32a708025cbb466cd08c99b98aL105-L109): Removed the addition of workflow-related sorting options within the `getFilterForm` method when the workflow feature is enabled, as this is now handled directly in the form definition.

Conditional option inclusion:

* [`libraries/src/Form/Field/ListField.php`](diffhunk://#diff-0b1716280a7421741ce1bafeec113db143676d193228e5c490efd3f7dde3f3fbR124-R128): Added a check to exclude options that require the workflow feature if the workflow feature is not enabled.


### Testing Instructions
Using Sample Data
Test 1. Enable workflows and observe in the sort filter dropdown the position of  Stage Asc/Desc in the list
Test 2. Disable Workflows and observe that Stage Asc/Desc is not present in the list


### Actual result BEFORE applying this Pull Request
1. At the bottom of the list
2. Not present

### Expected result AFTER applying this Pull Request
1. Near the top Directly below ordering to match the column order in the table
2. Not present

### Link to documentations
Please select:
